spog/ui/crates/components/src/cyclonedx/mod.rs

```diff
@@ -3,9 +3,16 @@ mod packages;
 pub use packages::*;
 
 use patternfly_yew::prelude::*;
+use serde_json::Value;
 use yew::prelude::*;
 
-pub fn cyclonedx_meta(bom: &cyclonedx_bom::prelude::Bom) -> Html {
+pub fn cyclonedx_meta(bom: &cyclonedx_bom::prelude::Bom, source: &str) -> Html {
+    let spec_version = serde_json::from_str::<Value>(source).ok().and_then(|json| {
```

Comment:
Parsing SBOMs can be a costly operation. I think that should be wrapped with `use_memo`.

Reply:
Sounds good. I just needed to convert the function to a `#[function_component()]` since `useMemo` only works on Function Components. My last commit should have the change